hs_composite_resource/models.py

```diff
@@ -384,6 +705,14 @@ def delete_old_xml_files(folder=''):
                 aggregation = self.get_aggregation_by_name(new_path)
                 delete_old_xml_files()
                 aggregation.create_aggregation_xml_documents()
+                # check if the affected aggregation is a model program aggregation
+                # then any associated model instance aggregation metadata needs to be set dirty
+                # in order to regenerate metadata xml files for these linked model instance aggregations
+                if aggregation.type_name() == "ModelProgramLogicalFile":
```

Comment:
This belongs in the create_aggregation_xml_documents method of ModelProgramLogicalFile.